doc/manual_installation/dovecot.rst

```diff
@@ -147,11 +147,11 @@ MySQL users
 
   default_pass_scheme = CRYPT
 
-  password_query = SELECT email AS user, password FROM core_user WHERE email='%u' and is_active=1
+  password_query = SELECT email AS user, password FROM core_user, admin_domain WHERE core_user.email='%Lu' AND core_user.is_active=1 AND admin_domain.name='%Ld' AND admin_domain.enabled=1
```

Comment:
Not at all, I'm not good at DBA. I'm running a server with about 150-200 simultaneous connections ans it seems fine.

Talking about optimization, maybe you could modify the password query so that user query is done at the same time. See http://wiki2.dovecot.org/UserDatabase/Prefetch